javascript/ql/src/semmle/javascript/GeneratedCode.qll

```diff
@@ -99,6 +99,21 @@ private predicate hasManyInvocations(TopLevel tl) {
   )
 }
 
+/**
+ * Holds if `f` is side effect free, and full of primitive literals, which is often a sign of generated data code.
+ */
+private predicate isData(File f) {
+  // heuristic: `f` has more than 1000 primitive literal expressions ...
+  count(SyntacticConstants::PrimitiveLiteralConstant e | e.getFile() = f) > 1000 and
+  // ... but no expressions with side effects ...
+  not exists (Expr e |
+    e.getFile() = f and
+    e.isImpure() and
+    // ... except for variable initializers
+    not e instanceof VariableDeclarator
```

Comment:
Does this mean we consider `VariableDeclarator`s to be impure? That seems undesirable.

Reply:
Yes, but that seems right to me. 
They do modify the scope object after all.

Reply:
That's true, but modelling it seems overkill, particularly since we don't even model the scope object. Also, that modification isn't observable in any way, is it?

Reply:
Hmm. I could go either way on this. On the one hand, shadowing variables in enclosing scopes and creation of properties on the global object are easily observable side effects, but on the other hand, I see what you mean regarding the expressive power of our analysis.
Do you want a change in this PR?

Reply:
You are right, of course, there are observable side effects, but I would still hesitate to ascribe them to the declarator, so on the whole I'd be in favour of changing this. It seems like a fairly minor and harmless change that would make this predicate look a bit less confusing.

Reply:
Done. 
Do we skip the full dist-compare for this change?

Reply:
Thanks. Yes, we do.